sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -92,6 +93,31 @@ object IntervalUtils {
   }
 
   private val yearMonthPattern = "^([+|-])?(\\d+)-(\\d+)$".r
+  private val yearMonthStringPattern =
+    "^(INTERVAL\\s+)([+|-])*?(')([+|-])?(\\d+)-(\\d+)(')(\\s+YEAR TO MONTH)$".r
+
+  def castStringToYMInterval(input: UTF8String): CalendarInterval = {
+    input.trimAll().toString.toUpperCase(Locale.ROOT) match {
+      case yearMonthPattern("-", year, month) =>
+        fromYearMonthString(s"-$year-$month")
```

Comment:
in `fromYearMonthString`, there is an inner method `toInterval`. Can we move it out and call it here?

Reply:
> in `fromYearMonthString`, there is an inner method `toInterval`. Can we move it out and call it here?

Done